ipapython/certdb.py

```diff
@@ -104,14 +105,18 @@ def __exit__(self, type, value, tb):
     def run_certutil(self, args, stdin=None, **kwargs):
         new_args = [CERTUTIL, "-d", self.secdir]
         new_args = new_args + args
+        if self.password_filename is not None:
+            new_args.extend(['-f', self.password_filename])
         return ipautil.run(new_args, stdin, **kwargs)
 
     def create_db(self, password_filename):
         """Create cert DB
 
         :param password_filename: Name of file containing the database password
         """
-        self.run_certutil(["-N", "-f", password_filename])
+        # run_certutil will use self.password_filename to setup the db
+        self.password_filename = password_filename
```

Comment:
Now that the filename is specified via `__init__()` argument, I see very little point in keeping the argument in `create_db()` as well.